src/FormikContext.tsx

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { FormikContext } from './types';
+
+const PrivateFormikContext = React.createContext<FormikContext<any>>({} as any);
+export const FormikProvider = PrivateFormikContext.Provider;
+export const FormikConsumer = PrivateFormikContext.Consumer;
+
+export function useFormikContext<Values>() {
+  return React.useContext<FormikContext<Values>>(PrivateFormikContext);
+}
```

Comment:
I'm not sure if this would just be adding more API without a good use case, but I wonder if it would be worth it to have a hooks only API, something like:

```js
const App = () => {
  const [formik] = useFormik();
  const [form] = useForm(formik);
  const [email] = useField("email", formik);
  const [password] = useField("password", formik);

  return (
    <form {...form}>
      <input {...email} />
      <input {...password} />
    </form>
  );
}
```
And then `FormikContext` just puts the values from `useFormik` on context for that use case.

Reply:
I don’t see this how this api is useful. You will end up having to prop drill.

Reply:
useForm is not a useful hook either IMHO

Reply:
99.999% of our `<Form>` components have 0 additional props, which to me means it’s a very good abstraction.

Reply:
my goal for useField is to make building your own Fieldset the goal. Eventually, perhaps even moving `<Field>` into its own package a la prop-types

Reply:
it goes without saying that your API would be faster and more perform at because it does not hook into context. however, what is the point of using hooks here aside from useFormik??? just make prop getters and spread?

Reply:
along those lines... this PR already has prop getters. So we technically already support this API by exporting useFormik()

Reply:
Yeah I agree with everything you said, it makes the API less usable.